src/strategies/properties/property.sml

```diff
@@ -138,41 +141,39 @@ fun breakUntilCharacter _ [] = ([],[])
            end;
 
 fun findAttributes s =
-    let fun ff (c::c'::L) = if (c,c') = (#":",#"{") then breakUntilCharacter #"}" L
-                           else let val (x,y) = ff (c'::L)
-                                in (x,c::y)
-                                end
-          | ff L = ([],L)
-        val (atts',rest) = ff (String.explode s)
-        val atts = Parser.splitStrip ";" (String.implode atts')
-    in (atts, String.implode rest)
-    end;
-
-(* assumes  we're not dealing with a pattern. It's easier this way *)
-fun findType s =
-    let val (_,s') = findAttributes s;
-        val (v,_,typ') = Parser.breakOn ":" s';
-        val typ = case typ' of "" => NONE | ts => SOME (Type.fromString ts)
-    in (typ, Label v)
-    end;
-
-fun fromKindValuePair (k,rawV) =
     let
-        val vs = stringOfValue rawV
-        val (atts,rest) = findAttributes vs;
-        val (xt,v) = if vs = "" then (NONE, Boolean true) else
-                     if k = Kind.Pattern then (NONE, Label rest) else
-                     if k = Kind.Type then (NONE, Type (Type.fromString rest))
-                     else findType rest;
-    in (k,v,xt,atts)
-    end;
+      val (v,_,As) = Parser.breakOn ":" s;
+      val A = Attribute.read As (* takes {sdf; fsdfd:=sdf ; fdsdf:=[type1 =>2. type2 => 1] ;fte} and returns a list of attributes*)
```

Comment:
Why does `Attribute.read` handle a whole collection of attributes, rather than a single attribute? I mentioned this earlier, see that comment.